doc/ref_arch/kubernetes/chapters/chapter01.md

```diff
@@ -5,21 +5,91 @@
-- Kubernetes-as-a-Service
+> to be written
+
+
+<a name="1.2"></a>
+### 1.2 Terminology
+
+|Term|Definition|Reference|
+|---|---|---|
+|Container Image|A container image is a lightweight, standalone, executable package of software that includes everything needed to run an application: code, runtime, system tools, system libraries and settings.|[Link](https://www.docker.com/resources/what-container)|
+|Container|A running instantiation of a container image.|-|
+|Pod|A Pod (as in a pod of whales or pea pod) is a group of one or more containers (such as Docker containers), with shared storage/network, and a specification for how to run the containers. A Pod’s contents are always co-located and co-scheduled, and run in a shared context.|[Link](https://kubernetes.io/docs/concepts/workloads/pods/pod/)|
+|Cluster|A cluster consists of at least one cluster master and multiple worker machines called nodes. These master and node machines run the Kubernetes cluster orchestration system.|[Link](https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-architecture)|
+|Master|The cluster master runs the Kubernetes control plane processes, including the Kubernetes API server, scheduler, and core resource controllers.|[Link](https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-architecture)|
+|Node|Nodes are the worker machines (physical or virtual) that run your containerised applications and other workloads. Each node is managed from the master, which receives updates on each node's self-reported status. A node runs the services necessary to support the Docker containers that make up your cluster's workloads. These include the Docker runtime and the Kubernetes node agent (kubelet) which communicates with the master and is responsible for starting and running Docker containers scheduled on that node.|[Link](https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-architecture)|
+|PaaS|The capability provided to the consumer is to deploy onto the cloud infrastructure consumer-created or acquired applications created using programming languages, libraries, services, and tools supported by the provider. The consumer does not manage or control the underlying cloud infrastructure including network, servers, operating systems, or storage, but has control over the deployed applications and possibly configuration settings for the application-hosting environment.|[NIST SP 800-145, The NIST Definition of Cloud Computing](https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-145.pdf)|
+|CaaS|Container-as-a-Service, a foundational application PaaS capability, in this case delivered as a Kubernetes cluster and other technologies such as CNI, Ingress, Service Mesh, LMA, etc. CaaS sits above VIM/NFVI and performs some application LCM based on instructions received from VNFM.|[Foundational PaaS Definition](http://www.theenterprisearchitect.eu/blog/categorize-compare-cloud-vendors/#row5column3)|
```

Comment:
- CaaS is not part of PaaS 
- CaaS can run on bare metal, not only on top of a VIM/NFVI
- From where are the list of these technologies are comming from?
@tomkivlin can you please clarify these?

Reply:
CaaS functionality of Kubernetes which is absolutely necessarry to LCM containers, while PaaS functionality of Kubernetes is everything on top of it.

Reply:
To resolve this and also discussion that follows later on CaaS, Paas, Iaas, etc. would it be worth creating a couple of high level diagrams for discussion purposes?

Reply:
PaaS vs CaaS was discussed at length at the meeting on Tuesday.  I would like to point people to the following article:
https://containerjournal.com/features/paas-vs-caas-wrong-question-ask/
I would agree with the author of this article (except his references to Docker, which we want to avoid).  In essence the article definitions are close to what was proposed by Gergely.  The article provides short definitions:
 1. PaaS – supports development & deployment but it happens on predefined platform.
 2. CaaS – supports packing & deployment but it allows for personalized platform.
We could work out our definitions based on these ideas, I suppose. If we agree to such ideas, I would suggest us limiting our K8s RA to CaaS, as we were inclined in the meeting; this is because I do not think that we want to start with any development frameworks (PaaS).  Any thoughts?

Reply:
I'd avoid the term PaaS as it's overloaded by now, because "platform" is too unspecific (platform for what?). Originally, it denoted app development platforms like Heroku and Google App Engine with their specific development tools, APIs, etc.

We should only focus on CaaS, but clarify CaaSes don't just run+LCM containers, but complex application stacks, too.

Reply:
I tend to agree with @fzdarsky. Yes, K8s is more than just a CaaS, however for simplicity's sake define CaaS as a container orchestration engine + bare minimums that is necessary to deploy a telco application. So in practice CaaS is:
- a K8s distro
- a Host OS
- a package manager (Helm)
- minimum set of standard networking and storage plugins

Reply:
I would propose adding some FCAPS to scope as well, at least logging and performance data management (still bare minimum)

Reasoning for logging: one could argue that the "cloud-native" app design for logging is just pushing things to standard out and error, and let the runtime handle it.
Logging related configuration resides in the container runtime (driver, rotation etc.), application has no control over it. So from this perspective at least gathering logs of the standard channels, and forwarding them somewhere for further processing or storing I would argue is very much within CaaS responsibility. Note: I'm not saying central collection, and analysis is, only gathering and forwarding

PM: basic core metric collection and store is actually done automatically by Kubelet, so it would be hard to pretend CaaS does not do PM :)
moreover K8s provides native API for scaling based on metrics. putting PM responsibility into PaaS layer would make "native" K8s horizontal scaling feature dependent on an optional PaaS layer

Reply:
I've removed PaaS and hopefully clarified what constitutes and CaaS and the CaaS Manager.  Will resolve this conversation - can raise new conversations based on latest commits.